### PR TITLE
Trunk-5482 OpenmrsUtil.conceptListHelper(..) to handle duplicates

### PR DESCRIPTION
**Pull request by Andu033**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5482?filter=-2

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by Andu033** [comment]
So this is the build with the tests added that fail in this condition of duplicates in descriptor.

**Comment by Andu033** [comment]
@mks-d  i did the test , highlighted the fail then resolved the todo , also in a manner that remove a lot of code duplicates . Can you review ? Thanks. :)

**Comment by Andu033** [comment]
I think that coverage decreased because i reduced the lines of code so the covered likes of code reduced also ( replaced 20 covered lines with 3 covered lines ) so the percentage decreased but the coverage is the same . Is this a problem ? How i should fix? Also the coverage for this file will increase after this PR of mine is merged :https://github.com/openmrs/openmrs-core/pull/2820 (+0.2% on the entire project).

**Comment by dkayiwa** [comment]
Can you squash the commits into one as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by Andu033** [comment]
@dkayiwa  ok . I did

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java [hunk 1]
```diff
@@ -812,39 +812,9 @@ public static int compareWithNullAsLatest(Date d1, Date d2) {
...
-			if (c != null) {
-				if (isSet) {
-					List<Concept> inSet = cs.getConceptsByConceptSet(c);
-					ret.addAll(inSet);
-				} else {
-					ret.add(c);
-				}
-			}
-		}
+		ret.addAll(conceptSetHelper(descriptor));
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/util/OpenmrsUtil.java]: You basically deprecated `conceptListHelper(String)` saying that it would just do what `conceptSetHelper(String)` does. So before we do that, we need to figure out what the original intent of both methods was. And then **add the appropriate JavaDoc**.

@dkayiwa do you remember by any chance what does methods were for?

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 2]
```diff
@@ -217,6 +219,15 @@ public void validatePassword_shouldFailWithoutUpperAndLowerCasePasswordIfNotAllo
 		OpenmrsUtil.validatePassword("admin", "test0nl1", "1-8");
 	}
 	
+	/**
+	 * @see OpenmrsUtil#conceptListHelper(String)
+	 */
+	@Test
+	public void conceptListHelper_shouldReturnListOfUniqueConceptsWithGivenNameInDescriptor() {
+		List<Concept> helperList = OpenmrsUtil.conceptListHelper("name:YES|name:YES");
+		assertTrue("The list should have just one element" , helperList.size() == 1);
```
Comment (mks-d) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java]: Better would be a test that also asserts what the expected list is made of, rather than just checking its size. There are many ways to have a size 1 array that is not the one we hope for.

I see that tests are simply missing basically for those two methods, they just didn't exist in the first place. So regular tests will have to be added too, not just the edge cases mentioned in the todo's.

Also those tests should be in `OpenmrsUtilUnitTest` with `ConceptService` being mocked. It is overkill to load the whole Spring Test context just to test the behaviour of such util methods (as you saw explained [here](https://talk.openmrs.org/t/a-quick-question-about-testing-some-modules/21547/9?u=mksd)).

Comment (Andu033) [reply api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java]: I wrote tests for them in this PR https://github.com/openmrs/openmrs-core/pull/2820. Should we finish this first ? Otherwise the other PR will be less relevant .

Comment (mks-d) [reply api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java]: @Andu033 yes, tests **and** JavaDoc.

Comment (Andu033) [reply api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java]: How to mock the ConceptService?
